.github/workflows/ci_tests.yml

```diff
@@ -27,40 +27,40 @@ jobs:
             python: 3.x
             toxenv: codestyle
 
-          - name: Python 3.9 with minimal dependencies
+          - name: Python 3.10 with minimal dependencies
             os: ubuntu-latest
-            python: 3.9
-            toxenv: py39-test
+            python: '3.10'
+            toxenv: py310-test
 
-          - name: Python 3.10 with all optional dependencies and coverage checking
+          - name: Python 3.11 with all optional dependencies and coverage checking
             os: ubuntu-latest
-            python: '3.10'
-            toxenv: py310-test-alldeps-cov
+            python: '3.11'
+            toxenv: py311-test-alldeps-cov
 
-          - name: macOS - Python 3.10 with all optional dependencies
+          - name: macOS - Python 3.11 with all optional dependencies
             os: macos-latest
-            python: '3.10'
-            toxenv: py310-test-alldeps
+            python: '3.11'
+            toxenv: py311-test-alldeps
 
-          - name: Windows - Python 3.10 with all optional dependencies
+          - name: Windows - Python 3.11 with all optional dependencies
             os: windows-latest
-            python: '3.10'
-            toxenv: py310-test-alldeps
+            python: '3.11'
+            toxenv: py311-test-alldeps
 
-          - name: Python 3.11 with all optional dependencies and pre-releases
+          - name: Python 3.12 with all optional dependencies and pre-releases
             os: ubuntu-latest
-            python: '3.11'
-            toxenv: py311-test-alldeps-predeps
+            python: '3.12'
+            toxenv: py312-test-alldeps-predeps
 
-          - name: Python 3.9 with oldest supported version of all dependencies
-            os: ubuntu-20.04
-            python: 3.9
-            toxenv: py39-test-oldestdeps
+          - name: Python 3.10 with oldest supported version of all dependencies
+            os: ubuntu-22.04
+            python: '3.10'
+            toxenv: py310-test-oldestdeps
 
-          # - name: Python 3.10 with latest dev versions of key dependencies
+          # - name: Python 3.11 with latest dev versions of key dependencies
           #   os: ubuntu-latest
-          #   python: '3.10'
-          #   toxenv: py310-test-devdeps
+          #   python: '3.11'
+          #   toxenv: py311-test-devdeps
```

Comment:
Why is this commented? Also you don't want py313 running in the matrix?

Reply:
I think that we copied this from a package template many years ago, and these were commented out. I uncommented them and switched them to Python 3.13.